subsys/shell/modules/date_service.c

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2020 Nick Ward
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stdlib.h>
+#include <shell/shell.h>
+#include <init.h>
+#include <string.h>
+#include <posix/time.h>
+#include <sys/timeutil.h>
+
+#define HELP_NONE "[none]"
+#define HELP_DATE_SET \
+	"<year: 1900-9999> <month: 1-12> <day: 1-31> " \
+	"<hour: 0-23> <minute: 0-59> <second: 0-60>"
+
+static void date_print(const struct shell *shell, struct tm *tm)
+{
+	shell_print(shell,
+		    "%d-%02d-%02d "
+		    "%02d:%02d:%02d UTC",
+		    tm->tm_year + 1900,
+		    tm->tm_mon + 1,
+		    tm->tm_mday,
+		    tm->tm_hour,
+		    tm->tm_min,
+		    tm->tm_sec);
+}
+
+static int cmd_date_set(const struct shell *shell, size_t argc, char **argv)
+{
+	struct tm tm = {0};
+	struct timespec tp;
+	int ret;
+
+	ARG_UNUSED(argc);
+
+	tm.tm_year = strtol(argv[1], NULL, 10);
+	if ((tm.tm_year < 1900) || (tm.tm_year > 9999)) {
```

Comment:
This is too strict; years before 1900 are allowed, and there is no documented limit on an upper bound [that I can find](https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/time.h.html).  Just accept any value that made it through `strtol`.

Reply:
Done